File: String/LongestPalindrome/LogtestPalindrome.cc
class LongestPalindrome {
	public:
		string longestPalindrome(string &s) {
			string result;
			
			if(s == null || s.size() == 0)
				return s;
			
			int n = s.size();
			int longest = 0, left = 0,  right = 0;
			
			for(int i = 0; i < n; ++i) {
				for(int j = i + 1; j < n; ++j) {
					string subStr = s.substr(i, j - i);
					if(isPalindrome(subStr) || subStr.size() > longest) {
						logest = j - i;
						left = i;
						right = j;
					}
					
				}
			}
			
			result = s.substr(left, right - left);
			return result;
		}

		
	private:
		bool isPalindrome(string &s) {
			if(s == null || s.size() == 0)
				return false;
			
			n = s.size();
			
			for(int i = 0; i < n; ++i) {
				if(s[i] != s[n - i - 1])
					return false;
			}
			return true;
		}
};

   

File: String/LongestPalindrome/LogtestPalindrome.py
class LongestPalindrome:
    def isLongestPalindrome(self, s):
        if not s:
            return ""
            
        n = len(s)
        longest, left, rigth = 0, 0, 0
        
        for i in xrange(0, n):
            for j in xrange(i + 1, n +1):
                substr = s[i:j]
                if self.isPalindrome(substr) and len(substr) > longest:
                    longest = len(substr)
                    left, right = i, j
                    
        result = s[left::right]
        return result
        
        
     def isPalindrome(self, s):
        if not s:
            return false
            
        return s == s[::-1]